validate/validators/sam.py

```diff
@@ -0,0 +1,24 @@
+'''Helper methods for SAM file validation'''
+from typing import Union
+from pathlib import Path
+import argparse
+
+import pysam
+
+def validate_sam_file(path):
+    '''Validates sam file'''
+    try:
+        pysam.quickcheck(str(path))
```

Comment:
I'm just curious to know the difference between SAM and BAM quickcheck.

Reply:
I haven't looked deeply into their code for any differences and their documentation wasn't super clear. I separated them here since the BAM check would also include an index check

Reply:
Got it. I was wondering if we could merge them but I guess not.